pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -108,27 +108,64 @@ func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	logger = logger.WithValues("deployment", dep.Name)
 
-	ing := &netv1.Ingress{}
-	ing.Name = util.FindOwnerKind(spc.OwnerReferences, "Ingress")
-	ing.Namespace = req.Namespace
-	logger = logger.WithValues("ingress", ing.Name)
-	if ing.Name != "" {
-		logger.Info("getting owner ingress")
-		if err = p.client.Get(ctx, client.ObjectKeyFromObject(ing), ing); err != nil {
+	return p.reconcileObjectDeployment(dep, spc, req, ctx, logger)
+}
+
+func (p *PlaceholderPodController) placeholderPodCleanCheck(obj client.Object) (bool, error) {
+	switch t := obj.(type) {
+	case *v1alpha1.NginxIngressController:
+		if t.Spec.DefaultSSLCertificate == nil || t.Spec.DefaultSSLCertificate.KeyVaultURI == nil {
+			return true, nil
+		}
+	case *netv1.Ingress:
+		managed, err := p.ingressManager.IsManaging(t)
+		if err != nil {
+			return false, fmt.Errorf("determining if ingress is managed: %w", err)
+		}
+		if t.Name == "" || t.Spec.IngressClassName == nil || !managed {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (p *PlaceholderPodController) reconcileObjectDeployment(dep *appsv1.Deployment, spc *secv1.SecretProviderClass, req ctrl.Request, ctx context.Context, logger logr.Logger) (ctrl.Result, error) {
+	var (
+		err error
+		obj client.Object
+	)
+
+	result := ctrl.Result{}
+
+	switch {
+	case util.FindOwnerKind(spc.OwnerReferences, "NginxIngressController") != "":
+		obj = &v1alpha1.NginxIngressController{}
+		obj.SetName(util.FindOwnerKind(spc.OwnerReferences, "NginxIngressController"))
+	case util.FindOwnerKind(spc.OwnerReferences, "Ingress") != "":
+		obj = &netv1.Ingress{}
+		obj.SetName(util.FindOwnerKind(spc.OwnerReferences, "Ingress"))
+		obj.SetNamespace(req.Namespace)
+	default:
+		p.events.Eventf(obj, "Waring", "FailedReconcileObjectDeployment", "failed to reconcile object deployment: object type not nginxIngressController or ingress")
```

Comment:
shouldn't warn. it's possible for user's to create their own spc which is completely fine and acceptable. just remove the event